src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -317,25 +326,65 @@ export class EventStoreComponent implements OnInit, OnDestroy {
     let parsedEvents : DataItem[] = [];
     for (const data of this.listEventStoreData) {
         if (data.eventsList.lastRefreshWasSuccessful) {
-            data.getEvents().forEach(event => parsedEvents.push(event));            
+            data.getEvents().forEach(event => parsedEvents.push(event));
         }
     }
+    
+    // create a new setting component
+    for (const data of this.listEventStoreData) {        
+        let visPresentFlag = false;
+        data.listSettings.columnSettings.forEach(setting => {
+            if (setting.propertyPath == "visPresent") {
+                visPresentFlag = true;
+            }
+        });
+        if (!visPresentFlag) {        
+            let newLogoSetting = new ListColumnSettingWithCustomComponent(
+                VisualizationLogoComponent,
+                "visPresent", 
+                "RCA Vis Present",
+                {
+                    enableFilter: true,
+                    colspan: 1
+                });  
+            newLogoSetting.fixedWidthPx = 100;
+            data.listSettings.columnSettings.splice(1, 0, newLogoSetting);
+            data.listSettings.secondRowColumnSettings.push(new ListColumnSettingWithCustomComponent(
+                VisualizationToolComponent,
+                '',
+                '',
+                {
+                    enableFilter: false,
+                    colspan: -1
+                }
+            ));
+        }        
+    }
 
-    // testing purposes    
-    this.testEvent(parsedEvents);
-  }
-
-  public findConcurrentEvents(): void {
-    const timelineEventSubs = this.listEventStoreData.map(data => data.eventsList.refresh());
-    forkJoin(timelineEventSubs).subscribe(() => {
-        this.getConcurrentEventsData();
-    });
-  }
+    // grab highcharts data for all events
+    for (let parsedEvent of parsedEvents) {
```

Comment:
im curious if there is a better solution for this. adding new properties to an object that doesnt have any type information for it is hard to maintain. Like would it be possible to save the state onto the event store component and pass a way to access the data to the customComponents.